prow/cmd/deck/main.go

```diff
@@ -116,6 +134,46 @@ func main() {
 		mux.Handle("/tide.js", gziphandler.GzipHandler(handleTide(configAgent, ta)))
 	}
 
+	if *userUrl != "" {
+		ua := &userAgent{
+			path: *userUrl,
+		}
+		mux.Handle("/user-data.js", gziphandler.GzipHandler(handleUserDashboard(ua)))
+	}
+
+	var appClient config.Client
+	if err := yaml.Unmarshal(bytes.TrimSpace(gitAppClientRaw), &appClient); err != nil {
+		logger.WithError(err).Fatal("Error unmarshalling git app client config.")
+	}
+	var cookieSecret config.Cookie
+	if err := yaml.Unmarshal(bytes.TrimSpace(cookieSecretRaw), &cookieSecret); err != nil {
+		logger.WithError(err).Fatal("Error unmarshalling cookie secret.")
+	}
+	decodedSecret, err := hex.DecodeString(cookieSecret.Secret)
+	if err != nil {
+		logger.WithError(err).Fatal("Error decoding cookie secret.")
+	}
+	cookie := sessions.NewCookieStore(decodedSecret)
+
+	gitOAuthConfig := &configAgent.Config().GitOAuthConfig
+	gitOAuthConfig.InitGitOAuthConfig(&appClient, cookie)
+
+	goa := gitoauth.NewGitOAuthAgent(gitOAuthConfig, logrus.WithField("client", "gitoauth"))
+	oauthClient := &oauth2.Config{
+		ClientID:     gitOAuthConfig.Client.ClientID,
+		ClientSecret: gitOAuthConfig.Client.ClientSecret,
+		RedirectURL:  gitOAuthConfig.RedirectURL,
+		Scopes:       gitOAuthConfig.Scopes,
+		Endpoint:     github.Endpoint,
+	}
+
+	userDashboardAgent := userdashboard.NewDashboardAgent(gitOAuthConfig, logrus.WithField("client", "user-dashboard"))
+	mux.Handle("/user-dashboard", userDashboardAgent.HandleUserDashboard(userDashboardAgent))
```

Comment:
IMO the user dashboard should live behind a feature flag, otherwise my next deployment is blocked until I figure out how to make it work for us.